### PR TITLE
DUOS-1097[risk=no]User Session Timeout is not redirecting to home

### PR DESCRIPTION
**Pull request by raejohanek**
Scope: 
- Wrap all axios methods (get, put, and post) if the call returns a 401 logging out the user and returning to the home page
- in fetchAny, check for 401 status and log out the user/return to the home page

Addresses:
https://broadworkbench.atlassian.net/browse/DUOS-1097

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] PR is labeled with a Jira ticket number and includes a link to the ticket
- [ ] PR is labeled with a security risk modifier [no, low, medium, high] 
- [ ] PR describes scope of changes

In all cases:

- [ ] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [ ] Get PO sign-off for all non-trivial UI or workflow changes
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/SignIn.js [hunk 1]
```diff
@@ -25,11 +25,11 @@ export const SignIn = hh(class SignIn extends Component {
       prev.clientId = clientKey;
       return prev;
     });
-  }
+  };
```
Comment (raejohanek) [inline src/components/SignIn.js:28]: address compilation warnings

**Review by JVThomas — CHANGES_REQUESTED** [review]
You shouldn't have to change any of the axios methods at all for this fix. The only component you need to change is ```AuthenticatedRoute.js```, especially now that the backend has a ```/me``` endpoint that you can use as a safe way to check is a session is active on the backend.

The idea is to redirect before the component even loads. The changes here would only happen after the fact.

**Comment by JVThomas** [comment]
As a starting point, look at ```Routes.js``` and ```AuthenticatedRoutes.js``` to get a high-level view of what the latter is trying to accomplish.

**Comment by rushtong** [comment]
One of the effects we're seeing is that sessions are timing out after the component loads, and when the user follows a rendered link. Those cases are causing 401's that aren't getting caught.

**Comment by JVThomas** [comment]
> One of the effects we're seeing is that sessions are timing out after the component loads, and when the user follows a rendered link. Those cases are causing 401's that aren't getting caught.

Oh, in that case look into axios interceptors
https://github.com/axios/axios#interceptors

Either way, you don't need a wrapper on every axios call. More likely than not third-party packages will have some sort of generic function or attribute that will allow you to define some sort of pre-processing step.

File: src/libs/ajax.js [hunk 2]
```diff
@@ -1101,6 +1114,10 @@ const fetchAny = async (...args) => {
   if (res.status >= 500) {
     await reportError(args[0], res.status);
   }
+  if (!res.ok && res.status === 401) {
```
Comment (raejohanek) [inline src/libs/ajax.js:1117]: Update fetchAny to include the same 401 check fetchOk includes

File: src/libs/ajax.js [hunk 3]
```diff
@@ -11,6 +11,19 @@ import axios from 'axios';
 import {DataUseTranslation} from './dataUseTranslation';
 import {isFileEmpty} from './utils';
 
+//define axios interceptor
+//to log out user and redirect to home when response has 401 status
+//return responses with statuses in the 200s and reject the rest
+axios.interceptors.response.use(function (response) {
+  return response;
+}, function (error) {
+  if (error.status === 401) {
```
Comment (JVThomas) [inline src/libs/ajax.js]: To get the status on an error, you'll need to do ```error.response.status```

**Review by JVThomas — CHANGES_REQUESTED** [review]

File: src/libs/ajax.js (hunk 3 above)
Comment (JVThomas) [reply src/libs/ajax.js]: It's specific to axios, which is probably why you handle errors differently here compared to fetchOk and fetchAny.

https://github.com/axios/axios#handling-errors

**Review by JVThomas — APPROVED** [review]
Code looks good. This should cover instances where consent requests happen after a component has loaded and after a user session has timed out.

**Pull request merged**